code/modules/mob/living/simple_animal/friendly/farm/traits/core.dm

```diff
@@ -0,0 +1,408 @@
+								T.times_eaten_from++
+								M.hunger += 150
+								if(T.times_eaten_from >= M.amount_eaten_carnivore)
+									M.visible_message("[M] finishes eating [T].")
+									T.gib()
+									T = null
+								return
+				else
+					if(T && T.stat != DEAD)
+						T.attack_animal(M)
+						M.do_attack_animation(T)
+					else if(T && T.stat == DEAD)
+						eating_animal = TRUE
+						attacking_animal = FALSE
+						M.visible_message("[M] takes a bite out of [T].")
+						T.times_eaten_from++
+						M.hunger += 150
+						if(T.times_eaten_from >= M.amount_eaten_carnivore)
+							M.visible_message("[M] finishes eating [T].")
+							T.gib()
+							T = null
+	attacking_animal = FALSE
+	eating_animal = FALSE
+	return
+
+/datum/farm_animal_trait/egg_layer
+	name = "Egg Layer"
+	description = "This animal will reproduce by breeding and then laying eggs."
+	manifest_probability = 0
+	continue_probability = 0
+	random_blacklist = 1
+	var/breeding_timer = 0
+	var/breed_max = 150
+	var/is_breeding = 0
+	var/is_laying_egg = 0
+	var/mob/living/simple_animal/farm/mate
+	var/obj/machinery/nest/nest
+
+/datum/farm_animal_trait/egg_layer/on_apply(var/mob/living/simple_animal/farm/M)
+	breed_max = rand(300,500)
+	return
+
+/datum/farm_animal_trait/egg_layer/on_priority_life(var/mob/living/simple_animal/farm/M)
+	if(M.stat)
+		return
+	if(M.gender != FEMALE)
+		return
+	if(M.child) // no
+		return // nee non nyet nien no nej never ever
+	if(is_breeding || is_laying_egg)
+		if(is_breeding)
+			if(!mate)
+				var/list/potential_mates = list()
+				for(var/mob/living/simple_animal/farm/F in oview(M,7))
+					if(F.child || F.stat)
+						continue
+					potential_mates += F
+				if(potential_mates.len)
+					mate = get_closest_atom(/mob/living/simple_animal/farm, potential_mates, M)
+			else
+				if(!M.Adjacent(mate))
+					walk_to(M, mate, 1)
+					sleep(30)
+					if(mate && M.Adjacent(mate))
+						M.visible_message("[M] and [mate] breed.")
+						is_breeding = 0
+						is_laying_egg = 1
+						for(var/datum/farm_animal_trait/T in M.dna.traits)
+							T.on_breed(M, mate)
+						return
+				else
+					if(mate && M.Adjacent(mate))
+						M.visible_message("[M] and [mate] breed.")
+						is_breeding = 0
+						is_laying_egg = 1
+						for(var/datum/farm_animal_trait/T in M.dna.traits)
+							T.on_breed(M, mate)
+						return
+	if(is_laying_egg)
+		if(!nest)
+			var/list/potential_nests = list()
+			for(var/obj/machinery/nest/F in oview(M,7))
+				potential_nests += F
+			if(potential_nests.len)
+				nest = get_closest_atom(/obj/machinery/nest, potential_nests, M)
+		else
+			if(M.loc != get_turf(nest))
+				walk_to(M, nest, 1)
+				sleep(30)
+				if(nest && M.loc == get_turf(nest))
+					M.visible_message("[M] lays an egg.")
+					var/obj/item/weapon/reagent_containers/food/snacks/egg/E = new M.egg_type(M.loc)
+					E.mother = M
+					E.father = mate
+					is_laying_egg = 0
+					for(var/datum/farm_animal_trait/T in M.dna.traits)
```

Comment:
Remie, are you okay? I'm starting to suspect the walls of your room have the words "ISTYPLESS-FORLOOP" and "WHYYYY" smeared on them in blood.

also full disclosure this made me laugh incredibly hard